docs/plugin_developers_guide/01_Fledge_plugins.rst

```diff
@@ -41,11 +41,109 @@ Plugins
 In this version of Fledge you have six types of plugins:
 
 - **South Plugins** - They are responsible for communication between Fledge and the sensors and actuators they support. Each instance of a Fledge South microservice will use a plugin for the actual communication to the sensors or actuators that that instance of the South microservice supports.
-- **North Plugins** - They are responsible for taking reading data passed to them from the South bound service and doing any necessary conversion to the data and providing the protocol to send that converted data to a north-side task.
 - **Storage Plugins** - They sit between the Storage microservice and the physical data storage mechanism that stores the Fledge configuration and readings data. Storage plugins differ from other plugins in that they are written exclusively in C/C++, however they share the same common attributes and entry points that the other filter must support.
-- **Filter Plugins** - Filter plugins are used to modify data as it flows through Fledge. Filter plugins may be combined into a set of ordered filters that are applied as a pipeline to either the south ingress service or the north egress task that sends data to external systems.
-- **Notification Rule Plugins** - These are used by the optional notification service in order to evaluate data that flows into the notification service to determine if a notification should be sent.
-- **Notification Delivery Plugins** - These plugins are used by the optional notification service to deliver a notification to a system when a notification rule has triggered. These plugins allow the mechanisms to deliver notifications to be extended.
+- **Filter Plugins** - Filter plugins are used to modify data as it flows through Fledge. One or more filter plugins may compose a pipeline which modifies, either, data flowing out from the South ingress tasks into the Fledge Storage system, or out from Fledge storage into the North egress tasks.
+- **Notification Rule Plugins** - Notification plugins evaluate data after it has been entered into the Fledge Storage, before it enters the northbound Filter pipeline(s). If the notification evaluates to True, a rule associated with the condition is executed. (Notification rules plugins require installation of the optional notification service.)
```

Comment:
Notification are not evaluated after data has been stored, it happens in parallel. 

"If the notification evaluates to True, a rule associated with the condition is executed." - no. If the notification rule evaluates fires then a notification delivery es executed. Rules can fire for triggering or clearing of the rule condition. 

"(Notification rules plugins require installation of the optional notification service.)"
->
"Notification rule plugins are applied to the notification service, the notification service is an optional service and must be installed before the notification plugins can be used."

Reply:
OK, mostly your text but how does this look: 

 **Notification plugins evaluate data in parallel with their entry into the Fledge Storage, before it enters the northbound Filter pipeline(s). If the notification "fires", notification delivery is executed. Rules can fire for triggering or clearing of the rule condition.
  Notification rule plugins are applied to the notification service, the notification service is an optional service and must be installed before the notification plugins can be used.**

What I've seen with the bad bearing is that it effects action by returning True from plugin eval. So, are you saying that the plugin_eval returns "deliver" as its true/false result, and that what's actually done isn't a "true" notification per se, just a signal to "do delivery"?  (And that it could signal that either after clearing its condition, when something needs to happen, etc., but that's a up to the delivery logic?)